docs/pages/developer-guide.md

```diff
@@ -936,6 +935,174 @@ In the example above, a server task can be accessed through the address:
+ST=CA
+C=US
+```
+
+Additional X.509 `Subject Alternative Names`, based on the pod `discovery` and `vip` confiagurations, are encoded into the certificate. If no `discovery` or port exposed over VIP is configured single the certificate comes with a single SAN.
+
+```
+DNSName([pod-index-task].[service-name].autoip.dcos.thisdcos.directory)
+```
+
+Each VIP exposed port will result in a new SAN entry:
+
+```
+DNSName([port-0-vip-prefix].[service-name].l4lb.thisdcos.directory)
+DNSName([port-1-vip-prefix].[service-name].l4lb.thisdcos.directory)
+```
+
+Providing a custom `prefix` under `discovery` level will result in replacing the default SAN `DNSName([pod-index-task].[service-name].autoip.dcos.thisdcos.directory)` with the configured name and pod index, i.e.:
+
+```
+DNSName([discovery-prefix-index].[service-name].autoip.dcos.thisdcos.directory)
+```
+
+This certificate configuration allows to run proper TLS hostname verification by client when task is accessed by one of [internal DNS names](https://docs.mesosphere.com/1.9/networking/dns-overview/).
+
+Each certificate is valid for **10 years**.
+
+### Artifacts format
+
+TLS artifacts can be provided to a task in two different formats:
+
+#### PEM
+
+Standard [PEM encoded](https://en.wikipedia.org/wiki/Privacy-enhanced_Electronic_Mail) certificate, private key and root CA certificate. Based on a file extension name it is possible to tell which artifact is in the file.
+
+A certificate file with **.crt** extension contains an end-entity certificate with possible chain of certificates leading to the root CA certificate. The root CA certificate is not part of end-entity certificate file.
+
+A **.key** file contains PEM encoded private key.
+
+A file with **.crt** extension contains root CA certificate without certificate chain.
+
+#### Java Keystore
+
+The [Java Keystore (JKS)](https://en.wikipedia.org/wiki/Keystore) is a repository of various certificates and private keys. When private key and certificate is requested a task will get 2 JKS format files.
+
+A **.keystore** is a JKS file that contains a private key with end-entity certificate and complete certificate chain including the root CA certificate. Certificate is stored under alias name **`default`**. The keystore and key are protected by single **`notsecure`** password.
+
+A **.truststore** is a JKS file that contains DC/OS root CA certificate stored as a trust certificate. The certificte is stored under alias **`dcos-root`**. The keystore is protected by **`notsecure`** password.
+
+The password **`notsecure`** that is protecting both JKS files (containing an end-entity certificate with private key and root CA certificate) is selected because most of Java tools and libraries validate for some password. It is not to meant to provide any additional protection. Security of both files is achieved by using DC/OS secrets store with file-based in-memory secrets. No other schedulers and tasks can access TLS artificates provisioned by a scheduler.
```

Comment:
that protects both

...has been selected because most Java tools and libraries require a password.

Both files are secured using the DC/OS secrets store...

No other schedulers or tasks can access...